src/crucible/aws/cloudwatch/anomaly_detector.clj

```diff
@@ -0,0 +1,45 @@
+(ns crucible.aws.cloudwatch.anomaly-detector
+  (:require [clojure.spec.alpha :as s]
+            [crucible.aws.cloudwatch :as cloudwatch]
+            [crucible.encoding.keys :refer [->key]]
+            [crucible.resources :refer [defresource spec-or-ref]]))
+
+(s/def ::value (spec-or-ref string?))
+
+(s/def ::name (spec-or-ref string?))
+
+(s/def ::dimension (s/keys :req [::name ::value]))
+
+(s/def ::dimensions (s/coll-of ::dimension :kind vector?))
```

Comment:
Does this need to be a vector? The JSON serializer will coerce non-map collections to an array anyway

Reply:
I have removed the requirement from the spec: https://github.com/brabster/crucible/pull/188/commits/73a771b0cbdefd96d6d4ca010eb8eaeccdd4321a